### DISCUSSION TITLE
odd Scripting Notification when working with PMSM model

### DISCUSSION THREAD

**Discussion by tburch-LSS** [OpenModelica/OpenModelica#13178] [Q&A] [answered]
i've been using the fundamental-wave PMSM model... 

Modelica.Magnetic.FundamentalWave.BasicMachines.SynchronousMachines.SM_PermanentMagnet

... and keep getting a strange Scripting Notification referring to one of its components (SinglePhaseElectroMagneticConverter). the model runs seemingly without problem. here is the notification:

[3] 14:27:50 Scripting Notification
Failed to deserialize json:  QJsonValue(object, QJsonObject({"$kind":"sub","exp":[{"$kind":"record","elements":[1,0],"name":"Complex"},{"$kind":"record","elements":[-0.5,0.866025],"name":"Complex"},{"$kind":"record","elements":[-0.5,-0.866025],"name":"Complex"}],"subscripts":["<singlePhaseElectroMagneticConverter, 1>"]}))
Expression: unsupported JSON object {
    "$kind": "sub",
    "exp": [
        {
            "$kind": "record",
            "elements": [
                1,
                0
            ],
            "name": "Complex"
        },
        {
            "$kind": "record",
            "elements": [
                -0.5,
                0.866025
            ],
            "name": "Complex"
        },
        {
            "$kind": "record",
            "elements": [
                -0.5,
                -0.866025
            ],
            "name": "Complex"
        }
    ],
    "subscripts": [
        "<singlePhaseElectroMagneticConverter, 1>"
    ]
} Failed to deserialize json:  QJsonValue(object, QJsonObject({"$kind":"sub","exp":[{"$kind":"record","elements":[1.22474,0],"name":"Complex"},{"$kind":"record","elements":[7.4994e-17,1.22474],"name":"Complex"}],"subscripts":["<singlePhaseElectroMagneticConverter, 1>"]}))
Expression: unsupported JSON object {
    "$kind": "sub",
    "exp": [
        {
            "$kind": "record",
            "elements": [
                1.22474,
                0
            ],
            "name": "Complex"
        },
        {
            "$kind": "record",
            "elements": [
                7.4994e-17,
                1.22474
            ],
            "name": "Complex"
        }
    ],
    "subscripts": [
        "<singlePhaseElectroMagneticConverter, 1>"
    ]
}

**Comment by tburch-LSS** [comment]
sorry, i forgot to point out: i'm using OpenModelica 1.24.1 on Ubuntu 22.04.5 LTS.

**Comment by perost** [comment, marked as answer]
Fixed with #13197.
